src/code/z_message.c

```diff
@@ -223,339 +807,4298 @@ void Message_FindMessage(PlayState* play, u16 textId) {
+    Player* player = GET_PLAYER(play);
+    f32 var_fv0;
+
+    if (play->msgCtx.msgMode == MSGMODE_NONE) {
+        gSaveContext.prevHudVisibility = gSaveContext.hudVisibility;
+    }
+
+    if (textId == 0xFF) {
+        func_80115844(play, DO_ACTION_STOP);
+        play->msgCtx.hudVisibility = gSaveContext.hudVisibility;
+        Interface_SetHudVisibility(HUD_VISIBILITY_A_B_C);
+        gSaveContext.save.unk_06 = 20;
+    } else if ((textId == 0x579) || (textId == 0x8D8)) {
+        Interface_SetHudVisibility(HUD_VISIBILITY_A_HEARTS_MAGIC_WITH_OVERWRITE);
+    } else if (((textId == 0x28) || (textId == 0x29) || (textId == 0x2A)) &&
+               (player->transformation == PLAYER_FORM_DEKU)) {
+        //! FAKE:
+        if (msgCtx) {}
+        textId = 0xC9;
+    } else if (textId == 0x11) {
+        if (gSaveContext.save.saveInfo.inventory.strayFairies[((void)0, gSaveContext.dungeonIndex)] == 0xF) {
+            textId = 0xF3;
+        }
+    } else if ((textId == 0x92) && (play->sceneId == SCENE_KOEPONARACE)) {
+        textId = 0xCD;
+    }
+
+    if ((textId == 0xC) && (GET_QUEST_HEART_PIECE_COUNT != 0)) {
+        textId = GET_QUEST_HEART_PIECE_COUNT;
+        textId += 0xC4;
+    }
+
+    msgCtx->currentTextId = textId;
+
+    if ((msgCtx->currentTextId == 0xC) || (msgCtx->currentTextId == 0xD) || (msgCtx->currentTextId == 0xC5) ||
+        (msgCtx->currentTextId == 0xC6) || (msgCtx->currentTextId == 0xC7)) {
+        Interface_SetHudVisibility(HUD_VISIBILITY_A_HEARTS_MAGIC_WITH_OVERWRITE);
+    }
+
+    msgCtx->messageHasSetSfx = false;
+    D_801C6A70 = 0;
+    msgCtx->textboxSkipped = false;
+    msgCtx->textIsCredits = false;
+    var_fv0 = 1.0f;
+
+    if (play->pauseCtx.bombersNotebookOpen) {
+        if (gSaveContext.options.language == LANGUAGE_JPN) {
+            msgCtx->textCharScale = 1.4f;
+            msgCtx->unk11FFC = 0x1E;
+            msgCtx->unk11FF8 = 0x32;
+            var_fv0 = 1.4;
+        } else {
+            msgCtx->textCharScale = 1.4f;
+            msgCtx->unk11FFC = 0x16;
+            msgCtx->unk11FF8 = 0x32;
+            var_fv0 = 1.4;
+        }
+    } else if (textId >= 0x4E20) {
+        msgCtx->textIsCredits = true;
+        msgCtx->textCharScale = 0.85f;
+        msgCtx->unk11FFC = 6;
+        msgCtx->unk11FF8 = 0x14;
+    } else if (gSaveContext.options.language == LANGUAGE_JPN) {
+        msgCtx->textCharScale = 0.88f;
+        msgCtx->unk11FFC = 0x12;
+        msgCtx->unk11FF8 = 0x32;
+    } else {
+        msgCtx->textCharScale = 0.75f;
+        msgCtx->unk11FFC = 0xC;
+        msgCtx->unk11FF8 = 0x41;
+    }
+
+    sCharTexSize = msgCtx->textCharScale * 16.0f;
+    sCharTexScale = 1024.0f / msgCtx->textCharScale;
+    D_801F6B08 = 1024.0f / var_fv0;
+
+    if (msgCtx->textIsCredits) {
+        Message_FindCreditsMessage(play, textId);
+        msgCtx->msgLength = font->messageEnd;
+        DmaMgr_SendRequest0(&font->msgBuf, &_staff_message_data_staticSegmentRomStart[font->messageStart],
```

Comment:
We may want to look at how to use the `SEGMENT_ROM_START` macro here, but it could be too annoying, so no need to do it right now

Reply:
Wasn't hard as there were already examples in z_message.